### PR TITLE
Improve `rcppmath` Doxygen documentation

### PR DESCRIPTION
**Pull request by aprotyas**
This PR broadly attempts to make `rcppmath` documentation more detailed, as well as fixes some Doxygen issues that previously prevented `rcppmath` documentation from building correctly using `rosdoc2`.

Specifically:
* Documented rolling mean accumulator methods.
* For the clamping methods:
  * Added `\sa` tag instead of explicitly asking user to "see the other function".
  * nit: Capitalization.

Signed-off-by: Abrar Rahman Protyasha <abrar@openrobotics.org>

### CODE REVIEW

File: include/rcppmath/rolling_mean_accumulator.hpp
Comment (clalancette) [inline include/rcppmath/rolling_mean_accumulator.hpp:30]: I know you didn't add this, but I'd honestly just remove this sentence.  This is an implementation detail, and doesn't tell us what the class is meant to do.

If we really want to keep it, then I'd lead with "Computes the mean of the last accumulated elements", and then somewhere in the extended description mention that it is a simplified version of boost rolling_mean.

**Review by clalancette — CHANGES_REQUESTED** [review]
I've got one thing to change here.  Otherwise it looks good to me.

File: include/rcppmath/rolling_mean_accumulator.hpp
Comment (aprotyas) [reply include/rcppmath/rolling_mean_accumulator.hpp:30]: I concur that this line should be changed. I'll remove the lengthy signature and mention boost rolling_mean in the extended description then.

Comment (aprotyas) [reply include/rcppmath/rolling_mean_accumulator.hpp]: Fixed in 6cd9a07.

File: include/rcppmath/rolling_mean_accumulator.hpp
Comment (clalancette) [inline include/rcppmath/rolling_mean_accumulator.hpp:32]:
```suggestion
 * written to avoid dragging in boost dependencies.
```

**Review by clalancette — APPROVED** [review]
One minor fixup to do.  I'll approve anyway, assuming you'll fix before merging.

**Comment by ros-discourse** [comment]
This pull request has been mentioned on **ROS Discourse**. There might be relevant details there:

https://discourse.ros.org/t/ros-2-tsc-meeting-minutes-2021-9-16/22372/1

**Pull request merged**
